Set Take2/ViewController.swift

```diff
@@ -58,56 +72,182 @@ class ViewController: UIViewController {
+                self.selectedPlayingCardViews.remove(at: cardIndex)
+                found = true
+            }
+        }
+    }
+    
+    func changeLayout(ofCard card : PlayingCardView) {
+        for cardView in self.cardViews {
+            if cardView.tag == card.tag {
+                let borderColor = cardView.contentView.layer.borderColor
+                if borderColor == UIColor.white.cgColor || borderColor == UIColor.orange.cgColor || borderColor == UIColor.red.cgColor {
+                    cardView.contentView.layer.borderWidth = 3.0
+                    cardView.contentView.layer.borderColor = UIColor.blue.cgColor
+                    cardView.contentView.layer.cornerRadius = 8.0
+                }
+                else if borderColor == UIColor.blue.cgColor {
+                    cardView.contentView.layer.borderWidth = 0
+                    cardView.contentView.layer.borderColor = UIColor.white.cgColor
+                    cardView.contentView.layer.cornerRadius = 0
+                }
+            }
+        }
+    }
+    
+    func handleThreeCardsSelected() {
+        checkIfSelectedCardsAreSet()
+        self.selectedPlayingCardViews.removeAll()
+        checkIfNeedToEndGame()
     }
     
     func startNewGame(){
         game = SetGame()
-        resetButtons()
-        selectedButtons.removeAll()
+        resetCardViews()
+        self.selectedPlayingCardViews.removeAll()
+        self.cardViews.removeAll()
         needToDealNewCards = false
         self.dealNewCardsButton.isEnabled = true
-        freeButtonIndex = 12
         initBoard()
         updateLabelsInUI()
     }
     
-    func resetButtons() {
-        for buttonIndex in 0..<self.buttons.count {
-            self.buttons[buttonIndex].layer.borderWidth = 0
-            self.buttons[buttonIndex].setAttributedTitle(NSAttributedString(string: ""), for: UIControlState.normal)
-            self.buttons[buttonIndex].layer.backgroundColor = UIColor.white.cgColor
-            self.buttons[buttonIndex].layer.cornerRadius = 0
+    func resetCardViews() {
+        for cardView in self.cardViews {
+            cardView.contentView.layer.borderWidth = 0
+            cardView.contentView.layer.borderColor = UIColor.white.cgColor
+            cardView.contentView.layer.cornerRadius = 0
         }
     }
     
     func initBoard() {
-        self.buttons.disableAllElements()
+        updateGrid()
+    }
+    
+    func updateGrid() {
         let cardsOnGameBoard = game.getCardsOnGameBoard()
+        let cardsGrid = Grid(layout: Grid.Layout.dimensions(rowCount: cardsOnGameBoard.count / 3, columnCount: 3), frame: CGRect(origin: CGPoint(x: 0, y: 0), size: self.boardView.frame.size))
         
-        if cardsOnGameBoard.count != game.getNumOfCardsOnStart() {
-            assertionFailure("Invalid number of cards on start")
+        for subView in self.boardView.subviews {
+            subView.removeFromSuperview()
```

Comment:
I would call `self.cardViews.removeAll()` here and this will allow you to remove lines 226-230

Reply:
I'm not quite sure why I need to remove lines 226-230. I need them in order to restore the properties of an existing view

Reply:
There is no need to restore if you create all the "card views" from scratch